### ISSUE TITLE
Workflows only being enabled once

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress-Future#956]
I'm testing a couple of the demo workflows
![Screenshot 2024-10-03 at 9 45 29 AM](https://github.com/user-attachments/assets/1102c8fd-3cc2-4e1f-8bb5-7ec07004ad50)

It seems possible to enable them for one post, but not a second post
![Screenshot 2024-10-03 at 9 45 22 AM](https://github.com/user-attachments/assets/d8464ce0-0688-4219-858e-c23c44e15364)

**Comment by andergmartins** [comment]
@stevejburge good catch. Actually we have a fix for this, but it is not implemented by default in the demo workflows. I will check if I can change that.

But please, could you try changing one the workflow "Unpublish to Draft after a week", editing the "Schedule" step, by adding `{{onSavePost1.post.ID}}` to the Unique ID Expression setting?

BTW, `onSavePost1` is the slug of the trigger, the step that provides the `post` variable, helping to differ each scheduled action. (It can be seem enabling the Advanced Settings).

![Screenshot 2024-10-03 at 15 31 05](https://github.com/user-attachments/assets/5d72e447-958d-46dd-a02d-ce1c0dcd75b2)

![Screenshot 2024-10-03 at 15 31 24](https://github.com/user-attachments/assets/60270522-5718-430f-ad12-8dc0a3ea423f)

**Comment by stevejburge** [comment]
@andergmartins Sorry, that doesn't seem to have solved it

**Comment by andergmartins** [comment]
Ok, thanks. I will double-check

**Comment by andergmartins** [comment]
@stevejburge ok, I think it is fixed for different posts in the same workflow. To avoid duplicate schedules after re-saving the same post, it has a flag that tells that a specific step has already run for that post (when, and how many times), and it will not run again if it is set as non-repeating. 
I'm not sure yet what is the best approach to keep the "non-repeating" control, but still allow to run the same action again if the user wants too. Since each workflow is unpredictable, it is hard to know when it should or not should run again for the post.

**Comment by andergmartins** [comment]
If we already had the "conditional" step, where the user can make the workflow take decisions, we could probably pass that responsibility to the user's hand instead.

**Comment by stevejburge** [comment]
Sounds great to me. Recurring workflows have been a common request

**Comment by andergmartins** [comment]
Ok, thanks. This moved #848 into this milestone.

**Comment by andergmartins** [comment]
Solved by adding a new option to the Schedule step to select the behavior to take when a duplicated task is found.

![image](https://github.com/user-attachments/assets/6e4c8df1-7524-442a-834a-a3563e36bdda)

**Comment by andergmartins** [comment]
@stevejburge ok, I see.

This is a difficult one. That limitation, or the need to specify a Unique ID came to solve the issue where 3rd party plugins we don't have any control can trigger the Wordpress hooks multiple times when a post is saved.

For example, it is common to see the `save_post` action being fired 2 or 3 times when a post is saved.

Imagine you have a workflow with the Post is Saved trigger, to send an email to the admin. Without that limitation it would schedule/send 3 emails.

We don't know what trigger, or how many triggers the user will have in the workflow so it is hard to predict what unique ID the workflow would need to create for the task in order to allow multiple schedules, but still avoid the issue of calls in sequence fired by 3rd party plugins.

I would love to remove the need for that, but I don't have a better fix yet.

I suggest we improve the description on this release, and try again to think on a better solution for the second question you commented on above.

**Comment by stevejburge** [comment]
Thanks @andergmartins 

In that case, can we:
- Move this setting up the screen so it's easier to find
- Expand the descriptions
